app/controllers/social_accounts_controller.rb

```diff
@@ -0,0 +1,67 @@
+class SocialAccountsController < ApplicationController
+  include SocialAccountsHelper
+  before_action :authorize, :set_social_account, only: [:show, :edit, :update, :destroy]
+  helper_method :generate_link,  :get_supported_social_networks
+
+  def new
+  end
+
+  def edit
+    @user = User.find(params[:user_id])
+    @social_account = @user.social_accounts.find(params[:id])
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @social_account = SocialAccount.new(social_account_params)
+    @social_account.user_id = @user.id
+    # If the account was created successfully we want to finish the request by redirecting to the edit user page.
+    if @social_account.save
+      # Redirect to setting since we only add social accounts there
+      redirect_to edit_user_path(@user)
+    else
+      # If the account was created unsuccessfully we want to render the users edit form but still be in the processing of
+      # the current request so that the user can correct her/his input.
+      render "users/edit"
+    end
+  end
+
+  def update
+    # TODO(ct): @user is nil in this point
+    if @social_account.update(social_account_params)
+      redirect_to edit_user_path(@social_account.user_id)
+    else
+      @user = User.find(@social_account.user_id)
+      render :edit
+    end
+  end
+
+  def index
+    # WORKAROUND because URL is wrong after creating an invalid account from
```

Comment:
Not sure why this is handled inside the index action. This should be handled inside the show action. If we are editing a social account (inside the edit view of the social account) and an error occurs we show the error message so that the user can correct the error. If the user sends manually from his browser a new request (for example by rerendering the page)  we need to handle that case.

Reply:
I removed the workaround. If we try to reload the page  (inside the browser) while editing the social account, we will stay at the edit page and the index action is doing nothing.